driver-sync/src/test/functional/com/mongodb/client/AbstractTransactionsTest.java

```diff
@@ -455,4 +486,54 @@ private boolean canRunTests() {
             return false;
         }
     }
+
+    private class TargetedFailPoint {
+        private final BsonDocument failPointDocument;
+        private final MongoDatabase adminDB;
+
+        TargetedFailPoint(final BsonDocument operation) {
+            final BsonDocument arguments = operation.getDocument("arguments", new BsonDocument());
+            final ClientSession clientSession = sessionsMap.get(arguments.getString("session").getValue());
+
+            if (clientSession.getPinnedMongosAddress() != null) {
+                MongoClientSettings.Builder builder = MongoClientSettings.builder()
+                        .applyToClusterSettings(new Block<ClusterSettings.Builder>() {
+                            @Override
+                            public void apply(final ClusterSettings.Builder builder) {
+                                builder.hosts(singletonList(clientSession.getPinnedMongosAddress()));
+                            }
+                        });
+                MongoClient mongoClient = MongoClients.create(MongoClientSettings.builder(builder.build())
+                        .applyToSocketSettings(new Block<SocketSettings.Builder>() {
+                            @Override
+                            public void apply(final SocketSettings.Builder builder) {
+                                builder.readTimeout(5, TimeUnit.SECONDS);
```

Comment:
Why is the read timeout necessary?  What happens if it's not set?

Reply:
Removed.